### PR TITLE
feat: reset etcd before every test run.

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

___
### New feature or improvement
- Describe the details and related test reports.
Failure of one test case may leave dirty data behind and cause other cases to fail.

___
### Please add the corresponding test cases if necessary.
Empty the etcd data before each test case to ensure a clean environment
___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Review by juzhiyuan — APPROVED** [review]

**Comment by juzhiyuan** [comment]
@LiteSun failed BE

**Comment by LiteSun** [comment]
> @LiteSun failed BE

OK

**Comment by nic-chen** [comment]
@LiteSun  please sync codes from branch master

**Comment by LiteSun** [comment]
> @LiteSun please sync codes from branch master

Got it.

**Review by nic-chen — COMMENTED** [review]
what would happen when test deleting?

File: web/cypress/support/index.js [hunk 1]
```diff
@@ -17,6 +17,14 @@
 /* eslint-disable no-undef */
 import './commands';
 import '@cypress/code-coverage/support';
+const { SERVE_ENV = 'dev' } = Cypress.env();
+
+before(() => {
+  // reset etcd before test
+  if (SERVE_ENV === 'test') {
+    cy.exec('etcdctl del --prefix /', { failOnNonZeroExit: false });
```
Comment (liuxiran) [inline web/cypress/support/index.js:25]: it seems that the prefix for apisix data stored in etcd is `/apisix`, so is it necessary to specify the prefix to `/apisix`insread of `/`?
cc @nic-chen

Comment (nic-chen) [reply web/cypress/support/index.js:25]: I think `/` is OK here for testing

Comment (liuxiran) [reply web/cypress/support/index.js:25]: got it

**Review by liuxiran — APPROVED** [review]

**Review by Jaycean — APPROVED** [review]
LGTM

**Comment by LiteSun** [comment]
> what would happen when test deleting?

Empty the etcd data before each test case to ensure a clean environment.

**Comment by juzhiyuan** [comment]
ping @nic-chen

**Comment by nic-chen** [comment]
> > what would happen when test deleting?
> 
> Empty the etcd data before each test case to ensure a clean environment.

I mean, if the ETCD data is cleared before each test, when test deleting a route, it has been cleared before the test starts, then the operation of deleting the route will fail.

**Comment by LiteSun** [comment]
> > > what would happen when test deleting?
> > 
> > 
> > Empty the etcd data before each test case to ensure a clean environment.
> 
> I mean, if the ETCD data is cleared before each test, when test deleting a route, it has been cleared before the test starts, then the operation of deleting the route will fail.

Well, creating and deleting in a test case does not clear the etcd after the route is created, it clears it before the entire test example starts, and this is done to provide a clean environment for testing.

**Review by nic-chen — APPROVED** [review]

**Pull request merged**